Library/Homebrew/cmd/fix-permissions.rb

```diff
@@ -0,0 +1,117 @@
+require "extend/ENV"
+
+module Homebrew
+  def fix_permissions
+    fix_permissions_shared(:wait => false)
+    system "/usr/bin/sudo", "-k"
+  end
+
+  ### !!!!!!!!!!
+  ### EVERYTHING between these markers is shared (and should be kept in sync)
+  ### with Homebrew/install/install
+  ### !!!!!!!!!!
+  def sudo *args
+    ohai "/usr/bin/sudo #{args.join ' '}"
+    system "/usr/bin/sudo", *args
+  end
+
+  def fix_permissions_shared(options={})
+    wait = options.fetch(:wait, true)
+
+    def chmod?(d)
+      File.directory?(d) && !(File.readable?(d) && File.writable?(d) && File.executable?(d))
+    end
+
+    def chown?(d)
+      File.directory?(d) && !File.owned?(d)
+    end
+
+    def chgrp?(d)
+      File.directory?(d) && !File.grpowned?(d)
+    end
+
+    def getc
+      system "/bin/stty raw -echo"
+      if STDIN.respond_to?(:getbyte)
+        STDIN.getbyte
+      else
+        STDIN.getc
+      end
+    ensure
+      system "/bin/stty -raw echo"
+    end
+
+    def wait_for_user
+      puts
+      puts "Press RETURN to continue or any other key to abort"
+      c = getc
+      # we test for \r and \n because some stuff does \r instead
+      abort unless c == 13 or c == 10
```

Comment:
Sorry, didn't want to detract from the actual discussion. I was just wondering whether we want to ignore style issues for new code (at least new in this repository), want to adjust the installer, or do something completely different. Wanted to raise this issue before the merge happens.